### PR TITLE
Put H5Web's dependencies as dependencies and externals of the app/lib packages

### PR DESCRIPTION
**Pull request by loichuder**

### CODE REVIEW

File: packages/webpack.config.js [hunk 1]
```diff
@@ -20,10 +23,9 @@ function getConfig(pkg) {
       filename: '[name].js',
       libraryTarget: 'commonjs2',
     },
-    externals: {
-      react: 'react',
-      'react-dom': 'react-dom',
-    },
+    externals: Object.fromEntries(
+      Object.keys(rootPkg.dependencies).map((dep) => [dep, `commonjs2 ${dep}`])
```
Comment (axelboc) [inline packages/webpack.config.js:27]: I mark all the dependencies as externals.

File: packages/webpack.config.js [hunk 2]
```diff
@@ -82,6 +84,18 @@ function getConfig(pkg) {
...
         ],
       }),
+      new JsonPostProcessPlugin({
+        matchers: [
+          {
+            // Add H5Web's dependencies to the package's output `package.json`
+            matcher: /^package\.json$/u,
+            action: (json) => ({
+              ...json,
+              dependencies: omit(rootPkg.dependencies, ['react', 'react-dom']),
```
Comment (axelboc) [inline packages/webpack.config.js:94]: I add all the dependencies to the `package.json` of `@h5web/lib` and `@h5web/app`. This is so we don't have to keep the version numbers in sync in three different `package.json` files...

**Review by axelboc — APPROVED** [review]

File: packages/webpack.config.js (hunk 2 above)
Comment (loichuder) [reply packages/webpack.config.js:94]: I see.

Could we use `.filter` to avoid the `lodash` dependency ?

Comment (axelboc) [reply packages/webpack.config.js:94]: Hmm it's quite convenient, and the dependency is only in `packages/package.json`, so it doesn't really matter.

**Comment by axelboc** [comment]
Closing in favour of moving to a monorepo structure.

**Pull request closed without merging**